transport/transporters/kafka/factory.go

```diff
@@ -0,0 +1,110 @@
+package kafka
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Nextdoor/pg-bifrost.git/shutdown"
+	"github.com/Nextdoor/pg-bifrost.git/stats"
+	"github.com/Nextdoor/pg-bifrost.git/transport"
+	"github.com/Nextdoor/pg-bifrost.git/transport/batch"
+	"github.com/Nextdoor/pg-bifrost.git/transport/transporters/kafka/transporter"
+	"github.com/Shopify/sarama"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/cevaris/ordered_map"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	logger = logrus.New()
+	log    = logger.WithField("package", "kafka")
+)
+
+func init() {
+	logger.SetOutput(os.Stdout)
+	logger.SetLevel(logrus.DebugLevel)
+}
+
+func New(
+	shutdownHandler shutdown.ShutdownHandler,
+	txnsWritten chan<- *ordered_map.OrderedMap,
+	statsChan chan stats.Stat,
+	workers int,
+	inputChans []<-chan transport.Batch,
+
+	transportConfig map[string]interface{}) []*transport.Transporter {
+
+	bootstrapHost := transportConfig[ConfVarBootstrapHost]
+	kafkaHost, ok := bootstrapHost.(string)
+	if !ok {
+		log.Fatalf("Expected type for %s is %s", ConfVarBootstrapHost, "string")
+	}
+
+	bootstrapPort := transportConfig[ConfVarBootstrapPort]
+	kafkaPort, ok := bootstrapPort.(string)
+	if !ok {
+		log.Fatalf("Expected type for %s is %s", ConfVarBootstrapPort, "string")
+	}
+
+	kafkaTopic := transportConfig[ConfVarTopic]
+	topic, ok := kafkaTopic.(string)
+	if !ok {
+		log.Fatalf("Expected type for %s is %s", ConfVarTopic, "string")
+	}
+	config, _ := producerConfig()
+	bootstrapServer := fmt.Sprintf("%s:%s", kafkaHost, kafkaPort)
+	syncProducer, _ := sarama.NewSyncProducer([]string{bootstrapServer}, config)
```

Comment:
Should there be a separate sync producer for each worker? Can we use the sync producer from different go routines?